code/procs/mobprocs/chatprocs.dm

```diff
@@ -91,48 +91,62 @@
 
 			src.say_verb(token + " " + text)
 
-	else if (src.ears && istype(src.ears, /obj/item/device/radio))
-		var/obj/item/device/radio/R = src.ears
-		if (R.bricked)
-			usr.show_text(R.bricked_msg, "red")
-			return
-		var/token = ""
-		var/list/choices = list()
-		choices += "[ headset_channel_lookup["[R.frequency]"] ? headset_channel_lookup["[R.frequency]"] : "???" ]: \[[format_frequency(R.frequency)]]"
+	else
+		var/has_headset = FALSE
+		var/mob/living/carbon/human/H = null
+		if ((src.ears && istype(src.ears, /obj/item/device/radio)))
+			has_headset = TRUE
+		else if (ishuman(src))	//Check if the decapitated skeleton head has a headset
+			H = src
+			if (isskeleton(H) && !H.organHolder.head && H.head_tracker.ears && istype(H.head_tracker.ears, /obj/item/device/radio))
+				has_headset = TRUE
+		if (has_headset)
+			var/obj/item/device/radio/R = null
+			if (H != null)
+				if (H.head_tracker != null && H.head_tracker.ears != null)
+					R = H.head_tracker.ears
+			else
+				R = src.ears
```

Comment:
I would just replace `var/has_headset = FALSE` with `var/obj/item/device/radio/R = null` on line 95, and then set `R` if we find a headset. And then just do `if (R)` on line 103 and skip lines 105-109 (since you already set `R`).